packages/components/text/src/text.styles.ts

```diff
@@ -122,6 +135,7 @@ export const bodyStyles = (props: TBodyProps) => css`
   ${props.isItalic && italic}
   ${props.isStrikethrough && strikethrough}
   ${props.tone && getTone(props.tone)}
+  ${props.fontWeight && getFontWeight(props.fontWeight)}
```

Comment:
I would suggest to merge the two props and remove the `${props.isBold && bold}` part

```suggestion
  ${getFontWeight(props)}
```

In the `getFontWeight` function then you can check for the `isBold` prop as well for backwards compatibility.

Reply:
@emmenko I get `Argument of type 'TBodyProps' is not assignable to parameter of type 'string'` when I try this. Am I doing something wrong?

Reply:
Prior the function is `const getFontWeight = (fontWeight: string) => string` so the argument `fontWeight` is a `string`. With the suggested change it's whatever props come in depending on what component it's called I presume. So the type needs to account for that as it's `TBodyProps`.

Reply:
@emmenko / @tdeekens 

Even when I fix the TS error it still breaks when I try and make this change. I tried doing something similar with getTone on line 137 and it broke too. Is there a reason to implement this prop differently than the pattern the others are using? Just trying to better understand :)

Reply:
I think the reasoning behind Nicola's proposal is to abstract away from consumers of the `getFontWeight` function the logic to know whether to use `isBold` or `fontWeight` properties.

Perhaps we can use an implementation like this one:
```ts
const getFontWeight = (props: { isBold?: boolean; fontWeight?: string; }) => {
  switch (props.fontWeight) {
    case 'regular':
      return `font-weight: ${designTokens.fontWeight400};`;
    case 'medium':
      return `font-weight: ${designTokens.fontWeight500};`;
    case 'bold':
      return `font-weight: ${designTokens.fontWeight600};`;
    default:
      if (props.isBold) {
        return `font-weight: ${designTokens.fontWeight600};`;
      }
      return ``;
  }
};
```

Then, we can update text components styles definitions like this:

**BEFORE**
```ts
export const bodyStyles = (props: TBodyProps) => css`
  ${getBaseStyles()}
  margin: 0;
  font-size: ${designTokens.fontSizeForTextAsBody};
  line-height: ${designTokens.lineHeightForTextAsBody};
  font-weight: ${designTokens.fontWeightForTextAsBody};
  ${props.isBold && bold}
  ${props.isItalic && italic}
  ${props.isStrikethrough && strikethrough}
  ${props.tone && getTone(props.tone)}
  ${props.fontWeight && getFontWeight(props)}
  ${props.truncate && truncate}
  ${props.nowrap && nowrap}
`;
```

**AFTER**
```ts
export const bodyStyles = (props: TBodyProps) => css`
  ${getBaseStyles()}
  margin: 0;
  font-size: ${designTokens.fontSizeForTextAsBody};
  line-height: ${designTokens.lineHeightForTextAsBody};
  font-weight: ${designTokens.fontWeightForTextAsBody};
  ${props.isItalic && italic}
  ${props.isStrikethrough && strikethrough}
  ${props.tone && getTone(props.tone)}
  ${getFontWeight(props)}
  ${props.truncate && truncate}
  ${props.nowrap && nowrap}
`;
```

Reply:
@CarlosCortizasCT makes sense, thanks for the explanation!